src/Compilers/Core/Portable/DiaSymReader/SymUnmanagedFactory.cs

```diff
@@ -154,6 +154,9 @@ private static Type GetComTypeType(ref Type lazyType, Guid clsid)
         {
             if (lazyType == null)
             {
+#if NET6_0_OR_GREATER
+                System.Diagnostics.Debug.Assert(OperatingSystem.IsWindows());
```

Comment:
@agocke why did you add this pattern in so many places?

Reply:
Sorry, missed this comment. This is a new analyzer in 6.0. The calls are annotated as "Windows only", so I asserted that we're in Windows to make the warnings go away. Flowing them upwards would also work, but it gets tricky.